app/controllers/outlets_controller.rb

```diff
@@ -1,30 +1,24 @@
- class OutletsController < ApplicationController
+class OutletsController < ApplicationController
 
-    def index
-      if params[:feed_data]
-        @outlets = Outlet.where("outlet_name > ?", params[:feed_data])
-      else
-        @outlets = Outlet.limit(5)
-      end
+  def index
+    @outlets = if params[:feed_data]
+                 Outlet.where(outlet_name: params[:feed_data])
+               else
+                 Outlet.limit(5)
+               end
 
-      respond_to do |format|
-        format.html
-        format.js
-      end
-    end
+    respond_to :html, :json
+  end
 
-    def show
-      @outlet = Outlet.find(params[:id])
-      if params[:feed_data]
-        @articles = @outlet.update_feed(params[:feed_data], 5)
-      else
-        @articles = @outlet.default_feed(10)
-      end
+  def show
+    @outlet = Outlet.find(params[:id])
+    @articles = if params[:feed_data]
+                  @outlet.update_feed(params[:feed_data], 5)
```

Comment:
Now that I think about it a little more.  You could just have a refresh_feed(params[:feed_date])
```suggestion
                  @outlet.refresh_feed(date: params[:feed_date])
```

then keep the logic in the model on how to proceed. 

```suggestion
                  def refresh_feed(date: nil)
                    if date
                     # logic for a refresh
                     else
                     # logic for a default. 
                     end
                     end
```

this would really help you in the end I think. I like to think of controllers as the dumb parts. They really only know how to pass data. they shouldn't be thinking about it if possible.